sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -213,12 +216,26 @@ def _on_outcome(self, outcome, condition):
                 :caption: Sends an event data and blocks until acknowledgement is received or operation times out.
 
         """
+        # Tracing code
+        parent_span = get_parent_span()
+        if parent_span:
+            child = parent_span.span(name="Azure.EventHubs.send")
+            child.kind = SpanKind.CLIENT  # Should be PRODUCER
+
+        def trace_message(message):
+            message_span = child.span(name="Azure.EventHubs.message")
+            message_span.start()
+            app_prop = dict(message.application_properties)
+            app_prop.setdefault(b"Diagnostic-Id", message_span.get_trace_parent().encode('ascii'))
```

Comment:
`get_trace_parent` return  the traceparent as a string, with format `"00-0af7651916cd43dd8448eb211c80319c-b9c7c989f97918e1-01"`. The encode makes it a byte string, and because we know there it's just 0-9A-F, `ascii` is fine.
So we send the bytes string `b"00-0af7651916cd43dd8448eb211c80319c-b9c7c989f97918e1-01"`